Version 1.1.0

pedropaulofb opened the pull request:
Please evaluate the proposel model for version 1.1.0.

All content was reviewed.

pedropaulofb committed:
add constraint and fix OntologyElement GS

pedropaulofb committed:
impreved documentation, added gitignore

pedropaulofb committed:
modifications 1 and 2

pedropaulofb committed:
added modifications 3 and 4

pedropaulofb committed:
add modification 5

pedropaulofb committed:
minor changes

pedropaulofb committed:
partial - up to test17

pedropaulofb committed:
partial improvements

pedropaulofb committed:
partial improvements

pedropaulofb committed:
final review

images re-generated, diagramation, etc.
improved documentation

claudenirmf committed:
Update README and .gitignore

claudenirmf committed:
Move diagrams

claudenirmf commented on images/Abstract Syntax.png:
- What is `{CP1}` in `Project`? The same goes for `Class`
- We need to change the datatype for multilingual text as discussed in the meeting.
- I do not agree that every `Relation` must have one stereotype. We do not have a complete set as of today. Have we discussed this?
- Let's discuss if we want to keep the `propertyType` as optional for `Property`. Maybe we want to raise some message of "incomplete" to the user in cases like this.
- Do not model `CollectiveClass` as a subtype of `Class`. Move the attribute to `Class` and make it optional.
- `Class` must have one or more values for `restrictedTo`

pedropaulofb replied:
> What is `{CP1}` in `Project`? The same goes for `Class`

They are constraints. We need to translate them to OCL. For now we have the following ones:

- `Project`: 
  - A Project cannot have a project relation to another Project. I.e., Projects are not in the domain list of the project relation.
- `Class`:
  - Every Class that is a type (i.e., has OntologicalNature with value 'type'), must have order equal or greater than two.
  - Every Class that is not a type (i.e., that has OntologicalNature != "type") has order = 1.
  - All classes with order one have their isPowertype attribute equal to false.

> We need to change the datatype for multilingual text as discussed in the meeting.

Done.

> Do not model `CollectiveClass` as a subtype of `Class`. Move the attribute to `Class` and make it optional.

Agreed. Implemented.

claudenirmf commented on images/Concrete Syntax.png:
- Just a curiosity: is there any difference between `Text` and `Rectangle` today? I guess there will be when we add support to text formatting.
- Another curiosity: is the association between `Shape` and `ElementView` an aggregation?

claudenirmf commented on images/Project.png:
- Review diagrams as you make changes (e.g., check if the additional attributes of `OntoumlElement` appeared in all diagrams. Also, keep the style and connectors placement consistent. Maybe we can consider reducing the number of diagrams to favor maintainability

pedropaulofb replied:
> Review diagrams as you make changes (e.g., check if the additional attributes of OntoumlElement appeared in all diagrams.

Done to `OntoumlElement` and to `Class`.

> Maybe we can consider reducing the number of diagrams to favor maintainability

I totally agree.

claudenirmf commented on images/Decoratable, Classifier, and Property.png:
- Maybe we should consider default value for the other attributes as well (e.g., `isDerived`), but these are heavily influenced by the stereotypes and nature restrictions.

pedropaulofb replied:
> * Maybe we should consider default value for the other attributes as well (e.g., `isDerived`), but these are heavily influenced by the stereotypes and nature restrictions.

For now, I added multiplicities to all attributes and set default values to all boolean ones.

I agree that some are influenced by stereotypes and nature restrictions. We can create constraints regarding that.

claudenirmf commented on images/ElementView.png:
- Let's consider how ternary relations will affect the abstract syntax. Right now, `RelationView` cannot accommodate this need.
- I am not sure that "diagram owner" is an OntoUML thing. This seems like a Visual Paradigm thing =/

claudenirmf commented on images/Classifier.png:
- If we keep this view, it might make sense to add property stereotypes to it

claudenirmf commented on images/Examples Class Diagram.png:
Example removed - unused.

claudenirmf commented on images/Metamodel Summary Diagram.png:
This color coding does not make sense outside the catalog paper

claudenirmf reviewed (commented):
Please keep track of the discussion points included for when we have a team meeting

claudenirmf committed:
Minor diagram updates

pedropaulofb committed:
update according to discussion

displayed attributes
removed MultiLingual Text
removed CollectiveClass
images re-generated

pedropaulofb committed:
review attributes

multiplicities added
default values added when boolean type
stereotype association fixed in one diagram

pedropaulofb commented:
Currently, we have the following "1st Level" classes:

- AggregationKind
- Cardinality
- OntologicalNature
- OntoumlElement
- Point
- Stereotype

The classes AggregationKind, OntologicalNature, and Stereotype are enumerations classes (extensional classes, defined by the set of their instances). So for me they are fine like this.

However, shouldn't we include Point as a OntoumlElement/DiagramElement and Cardinality as a OntoumlElement/ModelElement? IMHO, they are elements of these types and I see no reason to do not do that.

claudenirmf committed:
Update metamodel

claudenirmf committed:
Update metamodel
